src/runner/runner.py

```
@@ -0,0 +1,104 @@
+        """
+        self.logger = get_logger(__name__)
+        if idf:
+            self.idf = idf
+        else:
+            try:
+                IDF.setiddname(str(idd_file_path))
+                self.idf = IDF()
+            except Exception as e:
+                self.logger.error(
+                    f"Must provide either an IDF instance or a valid IDD file path. Error: {e}"
+                )
```

Ruff (0.14.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)